amqtt/broker.py

```diff
@@ -703,7 +714,7 @@ async def authenticate(self, session: Session, listener):
         # If all plugins returned True, authentication is success
         return auth_result
 
-    async def topic_filtering(self, session: Session, topic):
+    async def topic_filtering(self, session: Session, topic, action: str):
```

Comment:
I would prefer to see an `enum` defined for `action` - this way a type checker warns if an unknown action would be passed some place.

Reply:
Fair point, I'll make that amendment… I was initially thinking of using the KISS approach, but you're right, it's really an `Enum`.

Where's a good place to define the `Enum` itself?

Reply:
I would keep it inside the `broker.py`